libbeat/processors/add_process_metadata/config.go

```diff
@@ -35,6 +35,9 @@ type config struct {
 	// RestrictedFields make restricted fields available (i.e. env).
 	RestrictedFields bool `config:"restricted_fields"`
 
+	// IncludeCid make cid field available
+	IncludeCid bool `config:"include_cid"`
```

Comment:
There's already a way to select what fields you want added. See `Fields` which is below. Rather than add another option I think the enrichment should be triggered based on the presence of `container.id` in `Fields`. Note I chose `container.id` rather than `cid` because that's an ECS field ([source](https://www.elastic.co/guide/en/ecs/current/ecs-container.html)).

I would put `container.id` into the `defaultFields` so that it's enabled by default.